india_compliance/gst_india/report/gstr_1/gstr_1.py

```diff
@@ -933,6 +919,103 @@ def get_columns(self):
         self.columns = self.invoice_columns + self.tax_columns + self.other_columns
 
 
+class GSTR11A11BData:
+    def __init__(self, filters, gst_accounts):
+        self.filters = filters
+
+        self.pe = frappe.qb.DocType("Payment Entry")
+        self.pe_taxes = frappe.qb.DocType("Advance Taxes and Charges")
+        self.gl_entry = frappe.qb.DocType("GL Entry")
+        self.gst_accounts = gst_accounts
+
+    def get_data(self):
+        if self.filters.get("type_of_business") == "Advances":
+            return self.get_11A_data()
+
+        elif self.filters.get("type_of_business") == "Adjustment":
+            return self.get_11B_data()
+
+    def get_11A_data(self):
+        return (
+            self.get_query()
+            .select(
+                Sum(self.gl_entry.credit_in_account_currency).as_("cess_amount"),
+            )
+            .where(self.gl_entry.credit_in_account_currency > 0)
+            .groupby(
+                self.pe.place_of_supply,
+                self.gl_entry.account,
+                self.pe_taxes.rate,
+            )
+            .run(as_dict=True)
+        )
+
+    def get_11B_data(self):
+        query = (
+            self.get_query()
+            .select(
+                Sum(self.gl_entry.debit_in_account_currency).as_("cess_amount"),
+            )
+            .where(self.gl_entry.debit_in_account_currency > 0)
+            .groupby(
+                self.pe.place_of_supply,
+                self.gl_entry.account,
+                self.pe_taxes.rate,
+            )
+        )
+
+        return query.run(as_dict=True)
+
+    def get_query(self):
+        return (
+            frappe.qb.from_(self.gl_entry)
+            .join(self.pe)
+            .on(self.pe.name == self.gl_entry.voucher_no)
+            .join(self.pe_taxes)
+            .on(self.pe_taxes.parent == self.pe.name)
+            .select(
+                self.gl_entry.posting_date,
+                self.pe.name.as_("payment_entry"),
+                self.pe_taxes.rate,
```

Comment:
Case when. Don't take the rate from the cess account.
Should we sum rates so that we get CGST + SGST?

We are supporting more than 2 rows in the Taxes Table. Should we add validation in taxes table where more than 1 row for same account is used?

What if Actual is used instead of tax rate?

Better approach -->
- Total tax amount / Total Paid amount rounded up to 3 digits. ?
- Can we validated against `GST_TAX_RATES`